app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -0,0 +1,257 @@
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.selectedDate.collect {
+                    binding.date.text = it
+                }
+            }
+        }
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.selectedTime.collectLatest {
+                    binding.time.text = it
+                }
+            }
+        }
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.title.collectLatest {
+                    binding.taskTitle.text = it
+                }
+            }
+        }
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.description.collectLatest {
+                    binding.descriptionTextView.text = it
+                }
+            }
+        }
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.isInEditMode.collectLatest {
+                    setEditMode(it)
+                }
+            }
+        }
+    }
+
+    private fun setupAgendaItemType(){
+        binding.apply {
+            when (agendaItemType) {
+                AgendaItemType.TASK -> {
+                    taskColorBox.setImageResource(R.drawable.task_icon_box)
+                    taskType.text = getString(R.string.task)
+                }
+                AgendaItemType.EVENT -> {
+                    taskColorBox.setImageResource(R.drawable.event_icon_box)
+                    taskType.text = getString(R.string.event)
+                }
+                AgendaItemType.REMINDER -> {
+                    taskColorBox.setImageResource(R.drawable.reminder_icon_box)
+                    taskType.text = getString(R.string.reminder)
+                }
+            }
+        }
+    }
+
+    private fun onClickListeners(){
+        binding.apply {
+            closeButton.setOnClickListener {
+                navController.popBackStack()
+            }
+
+            editButton.setOnClickListener {
+                viewModel.setEditMode(true)
+            }
+
+            saveButton.setOnClickListener {
+                viewModel.setEditMode(false)
+            }
+
+            taskTitle.setOnClickListener {
+                editFragment(EditType.TITLE)
+            }
+
+            editTitleButton.setOnClickListener {
+                editFragment(EditType.TITLE)
+            }
+
+            editDescriptionButton.setOnClickListener {
+                editFragment(EditType.DESCRIPTION)
+            }
+
+            descriptionTextView.setOnClickListener {
+                editFragment(EditType.DESCRIPTION)
```

Comment:
Similar as for `showTimePickerFragment()` I'd recommend calling this something like `navigateToEditFragment()`